File: newDemo/newDemo/Base/View/BaseView.m
//
//  BaseView.m
//  BaseToolSystem
//
//  Created by 靳建南 on 2018/11/6.
//  Copyright © 2018年 靳建南. All rights reserved.
//

#import "BaseView.h"

@implementation BaseView

#pragma mark - 通过响应者链返回view的controller
- (UIViewController *)viewController:(UIView *)view{
    //将view放入响应者链中
    UIResponder *responder = view;
    //响应者链把事件不断上传
    while ((responder = [responder nextResponder]))
        //当传至viewController时，返回viewController
        if ([responder isKindOfClass: [UIViewController class]])
            return (UIViewController *)responder;
    
    return nil;
}
-(void)showHud:(NSString *)title animated:(BOOL)animated afterHiden:(NSTimeInterval)time
{
    _hud = [MBProgressHUD showHUDAddedTo:self animated:YES];
    
    //    if(title.length > 0) {
    //        _hud.mode = MBProgressHUDModeText;
    //        _hud.animationType = MBProgressHUDAnimationZoomIn;
    //        _hud.label.text = title;
    //        _hud.removeFromSuperViewOnHide = animated;
    //        [_hud hideAnimated:YES afterDelay:time];
    //    }
    _hud.mode = MBProgressHUDModeText;
    _hud.animationType = MBProgressHUDAnimationZoomIn;
    _hud.label.text = title;
    _hud.label.font = [UIFont systemFontOfSize:13];
    _hud.label.numberOfLines = 0;
    _hud.removeFromSuperViewOnHide = animated;
    [_hud hideAnimated:YES afterDelay:time];
    
}
-(void)showHudImgName:(NSString *)imgName animated:(BOOL)animated afterHiden:(NSTimeInterval)time
{
    _hud = [MBProgressHUD showHUDAddedTo:self animated:YES];
    
    _hud.mode = MBProgressHUDModeIndeterminate;
    //    _hud.mode = MBProgressHUDModeCustomView;
    //    _hud.customView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:imgName]];
    _hud.animationType = MBProgressHUDAnimationZoomIn;
    _hud.removeFromSuperViewOnHide = animated;
    [_hud hideAnimated:YES afterDelay:time];
    
}
-(void)hideHud{
    [_hud hideAnimated:YES];
}
-(NSString *)setTockenOfModule:(NSString *)module andController:(NSString *)controller andAction:(NSString *)action{
    
    //当前时间
    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
    formatter.dateFormat = @"yyyy-MM-dd";
    NSString *dateStr = [formatter stringFromDate:[NSDate date]];
    //拼接tocken
    NSString *md5Str = [NSString stringWithFormat:@"%@@%@@%@@%@",module,controller,action,dateStr];
    //md5加密
    NSString *token = [MD5Encryption md5:md5Str];
    return token;
}
//获取请求头的tocken
-(NSString *)setHeaderTockenOfModule:(NSString *)module andController:(NSString *)controller andAction:(NSString *)action andCurretnTime:(NSString *)currentTime{
    
    //拼接tocken
    NSString *md5Str = [NSString stringWithFormat:@"%@%@%@%@",module,controller,action,currentTime];
    //md5加密
    NSString *token = [MD5Encryption md5:md5Str];
    return token;
}

@end

File: BaseToolClass/BaseToolClass/Base(基类)/Main/BaseView.swift
//
//  BaseView.swift
//  BaseToolClass
//
//  Created by 靳建南 on 2018/11/9.
//  Copyright © 2018年 靳建南. All rights reserved.
// 

import UIKit

class BaseView: UIView {

    var hud:MBProgressHUD!
    
    //view跳转vc
    func viewController(view:UIView) -> UIViewController?{
        var next:UIResponder?
        next = self.next!
        repeat {
            //2.判断响应者对象是否是视图控制器类型
            if ((next as?UIViewController) != nil) {
                return (next as! UIViewController)
                
            }else {
                next = next?.next
            }
            
        } while next != nil
        
        return UIViewController()
    }
    
    //HUD加载提示
    func showHud(title:String,animated:Bool,time:TimeInterval){
        
        hud = MBProgressHUD.showAdded(to: self, animated: true)
        hud.mode = .text
        hud.animationType = .zoomIn
        hud.label.text = title;
        hud.label.font = UIFont.systemFont(ofSize: 14)
        hud.label.numberOfLines = 0
        hud.removeFromSuperViewOnHide = animated
        hud.hide(animated: true, afterDelay: time)
    }
    //显示图片
    func showHudImgName(imgName:String,animated:Bool,time:TimeInterval){
        hud = MBProgressHUD.showAdded(to: self, animated: true)
        hud.mode = .indeterminate
        hud.animationType = .zoomIn
        hud.removeFromSuperViewOnHide = animated
        hud.hide(animated: true, afterDelay: time)
    }
    //隐藏HUD
    func hideHud(){
        hud.hide(animated: true)
    }

}
